bitmex-stream/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+use async_stream::stream;
+use futures::SinkExt;
+use futures::Stream;
+use futures::StreamExt;
+use std::time::Duration;
+use tokio_tungstenite::tungstenite;
+
+pub use tokio_tungstenite::tungstenite::Error;
+
+/// Connects to the BitMex websocket API, subscribes to the specified topics (comma-separated) and
+/// yields all messages.
+///
+/// To keep the connection alive, a websocket `Ping` will be sent every 5 seconds in case no other
+/// message was received in-between. This is according to BitMex's API documentation: https://www.bitmex.com/app/wsAPI#Heartbeats
+pub fn subscribe(
+    topics: impl AsRef<str> + Send + 'static,
```

Comment:
❓ Why not a list of strings?

Reply:
Could be a list yeah! I iterated on that API a lot, had a brief thought of making it a list but opted not to because we don't have a use for it at the moment.

You also thinking of it might be a good enough reason to do it though!

Reply:
@luckysori What do you think of the last commit? Worth the extra dependency or should I just go with using `String` and require other callsites to use `to_owned` if they can't produce a `&'static str`?

Reply:
I like the fixup commit! It's true that dependency is not gonna be used much, but if it doesn't have a significant effect on our dependency graph then I'm okay with it.

Reply:
It is another proc-macro that has be compiled before all other dependencies so I think I am gonna go with the allocation instead and remove it again.